Assets/MixedRealityToolkit.Services/InputSystem/GazeProvider.cs

```diff
@@ -521,6 +521,25 @@ private async void RaiseSourceDetected()
             GazePointer.BaseCursor?.SetVisibility(true);
         }
 
+        internal void UpdateGazeInfoFromHit(MixedRealityRaycastHit raycastHit)
+        {
+            HitInfo = raycastHit;
+            if (raycastHit.transform != null)
+            {
+                GazeTarget = raycastHit.transform.gameObject;
+                var ray = GazePointer.Rays[0];
+                var lhd = (raycastHit.point - ray.Origin).magnitude;
```

Comment:
> var lhd = (raycastHit.point - ray.Origin).magnitude; [](start = 15, length = 53)

Do we also need to update gazeProvider.lastHitDistance?

See lines 266-268.

More generally, it might be worth having a helper function that takes an object and properly sets all three things, so that we don't forget to update them all as a unit (otherwise we risk similar bugs in the future where we only update 2/3 or 1/3 of the things)

Reply:
Refactored, but forgot to fix lastHitDistance, so leaving open.